ui/src/app/views/concept-homepage/component.spec.ts

```diff
@@ -71,12 +72,89 @@ describe('ConceptHomepageComponent', () => {
         { provide: ActivatedRoute, useValue: activatedRouteStub }
       ]}).compileComponents().then(() => {
         fixture = TestBed.createComponent(ConceptHomepageComponent);
+        // This tick initializes the component.
         tick();
+        // This finishes the API calls.
+        updateAndTick(fixture);
+        // This finishes the page reloading.
+        updateAndTick(fixture);
       });
   }));
 
 
   it('should render.', fakeAsync(() => {
     expect(fixture).toBeTruthy();
   }));
+
+  it('should have one card per domain.', fakeAsync(() => {
+    expect(fixture.debugElement.queryAll(By.css('.card.item-card')).length)
+      .toBe(DomainStubVariables.STUB_DOMAINS.length);
+  }));
+
+  it('should default to standard concepts only, and performs a full search', fakeAsync(() => {
+    const spy = spyOn(TestBed.get(ConceptsService), 'searchConcepts')
+      .and.callThrough();
+
+    const searchTerm = 'test';
+
+    simulateInput(fixture, fixture.debugElement.query(By.css('#concept-search-input')), searchTerm);
+    simulateClick(fixture, fixture.debugElement.query(By.css('.btn-search')));
+    updateAndTick(fixture);
+
+    DomainStubVariables.STUB_DOMAINS.forEach((domain) => {
+      const expectedRequest = {
+        query: searchTerm,
+        // Tests that it searches only standard concepts.
+        standardConceptFilter: StandardConceptFilter.STANDARDCONCEPTS,
+        domain: domain.domain
+      };
+      expect(spy).toHaveBeenCalledWith(
+        WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
+        WorkspaceStubVariables.DEFAULT_WORKSPACE_ID,
+        expectedRequest);
+    });
+    // Tests that it makes a call for each domain.
+    expect(spy).toHaveBeenCalledTimes(DomainStubVariables.STUB_DOMAINS.length);
```

Comment:
I like this verification. I think this is a good usage of a spy e.g. ensure we didn't make an excess number of calls.